File: src/sample/Velocity.java
package sample;

public class Velocity {

    private double velocityX = 0;
    private double velocityY = 0;
    private double sprintFactor;
    private double baseVelocity;


    public Velocity(double sprintFactor, double baseVelocity) {
        this.sprintFactor = sprintFactor;
        this.baseVelocity = baseVelocity;
    }

    public void setVelocityYPos() {
        velocityY = baseVelocity;
    }

    public void setVelocityXPos() {
        velocityX = baseVelocity;
    }

    public void setVelocityYNeg() {
        velocityY = -baseVelocity;
    }

    public void setVelocityXNeg() {
        velocityX = -baseVelocity;
    }

    public void resetVelocityY() {
        velocityY = 0;
    }

    public void resetVelocityX() {
        velocityX = 0;
    }

    public double getVelocityX() {
        return velocityX;
    }

    public double getVelocityY() {
        return velocityY;
    }

    public void clearVelocities() {
        resetVelocityX();
        resetVelocityY();
    }

    public void setRunning() {
        velocityX *= sprintFactor;
        velocityY *= sprintFactor;
    }
}

File: src/main/java/game/Entity/Velocity.kt
package game.Entity

class Velocity(private val sprintFactor: Double, private val baseVelocity: Double) {

    var velocityX = 0.0
        private set

    var velocityY = 0.0
        private set

    fun setVelocityYPos() {
        velocityY = baseVelocity
    }

    fun setVelocityXPos() {
        velocityX = baseVelocity
    }

    fun setVelocityYNeg() {
        velocityY = -baseVelocity
    }

    fun setVelocityXNeg() {
        velocityX = -baseVelocity
    }

    private fun resetVelocityY() {
        velocityY = 0.0
    }

    private fun resetVelocityX() {
        velocityX = 0.0
    }

    fun clearVelocities() {
        resetVelocityX()
        resetVelocityY()
    }

    fun setRunning() {
        velocityX *= sprintFactor
        velocityY *= sprintFactor
    }
}
